File: Entregavel 2/2 - player_inventory.py
"""Descrição: refatore a classe Jogador com todos os atributos 
privados e validados 
Implemente propriedades para hp (não pode exceder 
hp máximo) 
□Criar método receber_dano( ) com validação 
□Criar método curar( ) com validação 
□Garantir que nível nunca seja negativo
"""

class Jogador:
    def __init__(self, nome, nivel, hp, lvl_combate, lvl_magia, lvl_alquimia, lvl_habilidade_geral):
        self.__nome = nome

        if(self.__valida_atributo_construtor(nivel) == True):
            self.__nivel = nivel
        else:
            self.__nivel = 0

        if(self.__valida_atributo_construtor(hp) == True):
            self.__hp = hp
        else:
            self.__hp = 10

        if(self.__valida_atributo_construtor(lvl_combate) == True):
            self.__lvl_combate = lvl_combate
        else:
            self.__lvl_combate = 0

        if(self.__valida_atributo_construtor(lvl_magia) == True):
            self.__lvl_magia = lvl_magia
        else:
            self.__lvl_magia = 0

        if(self.__valida_atributo_construtor(lvl_alquimia) == True):
            self.__lvl_alquimia = lvl_alquimia
        else:
            self.__lvl_alquimia = 0

        if(self.__valida_atributo_construtor(lvl_habilidade_geral) == True):
            self.__lvl_habilidade_geral = lvl_habilidade_geral
        else:
            self.__lvl_habilidade_geral = 0

        self.__hp_maximo = 999

    def __valida_atributo_construtor(self, valor):
        retorno = False

        if(valor > 0):
            retorno = True

        return retorno

    def __valida_dano(self, valor_dano):
        retorno = False

        vida_total = self.__hp - valor_dano

        if(vida_total >= 0):
            retorno = True

        return retorno

    def __valida_cura(self, valor_cura):
        retorno = False

        vida_total = self.__hp + valor_cura

        if(vida_total <= self.__hp_maximo):
            retorno = True

        return retorno

    def recebe_dano(self, valor):
        if(self.__valida_dano(valor) == True):
            self.__hp = self.__hp - valor

    def recebe_cura(self, valor):
        if(self.__valida_cura(valor) == True):
            self.__hp = self.__hp + valor


    def __str__(self):
        return f"|{self.__nome}|HP:{self.__hp}|LVL:{self.__nivel}|Alquimia:{self.__lvl_alquimia}|Combate:{self.__lvl_combate}|Geral:{self.__lvl_habilidade_geral}|Magia:{self.__lvl_magia}|"
    

if __name__ == "__main__":

    jogador1 = Jogador("Geralt", 100, 500, 80, 60, 40, 70)

    print(jogador1)
    jogador1.recebe_dano(100)
    print(jogador1)
    jogador1.recebe_cura(50)

    print("-" * 68)

    jogador2 = Jogador("Yennefer", 90, 400, 30, 95, 50, 60)
    print(jogador2)
    jogador2.recebe_dano(450)  
    print(jogador2)

    print("-" * 68)

    jogador3 = Jogador("Ciri", 120, 450, 90, 40, 30, 80)
    print(jogador3)
    jogador3.recebe_cura(1000) 
    print(jogador3)

    print("-" * 68)

File: Entregavel 2/2 - player_inventory.cpp
/* Descrição: refatore a classe Jogador com todos os atributos 
privados e validados 
Implemente propriedades para hp (não pode exceder 
hp máximo) 
□Criar método receber_dano( ) com validação 
□Criar método curar( ) com validação 
□Garantir que nível nunca seja negativo */

#include <string.h>
#include <iostream>
using namespace std;

class Jogador{

    private:
        string nome;
        int nivel;
        int hp;
        int hp_maximo = 999;
        int lvl_combate;
        int lvl_magia;
        int lvl_alquimia;
        int lvl_habilidade_geral;

        bool valida_atributo_construtor(int valor){

            bool retorno = false;

            if(valor > 0){

                retorno = true;

            }

            return retorno;

        }

        bool valida_dano(int valor_dano){

            int vida_total = hp - valor_dano;

            bool retorno = false;

            if(vida_total >= 0){

                retorno = true;

            }

            return retorno;

        }

        bool valida_cura(int valor_cura){

            int vida_total = hp + valor_cura;

            bool retorno = false;

            if(vida_total <= hp_maximo){

                retorno = true;

            }

            return retorno;

        }

    public:

        Jogador(string nm, int lvl, int hp_user, int lvl_comb, int lvl_mag, int lvl_al, int lvl_ge){

            nome = nm;

            if(valida_atributo_construtor(lvl)== true){

                nivel = lvl;

            }else {

                nivel = 0;
            }

            if(valida_atributo_construtor(hp_user) == true){

                hp = hp_user;

            }else{

                hp = 10;

            }


            if(valida_atributo_construtor(lvl_comb) == true){

                lvl_combate = lvl_comb;

            }else{

                lvl_combate = 0;

            }

            if(valida_atributo_construtor(lvl_al) == true){

                lvl_alquimia = lvl_al;

            }else{

                lvl_alquimia = 0;

            }

            if(valida_atributo_construtor(lvl_mag) == true){

                lvl_magia = lvl_mag;

            }else{

                lvl_magia = 0;

            }

            if(valida_atributo_construtor(lvl_ge) == true){

                lvl_habilidade_geral = lvl_ge;

            }else{

                lvl_habilidade_geral = 0;
            }

        }

        void recebe_dano(int valor){

            if(valida_dano(valor) == true){

                hp = hp - valor;

            }else{

                cout << "Dano invalido" << endl;

            }


        }

        void recebe_cura(int valor){

            if(valida_cura(valor) == true){

                hp = hp + valor;

            }else{

                cout << "Cura invalida" << endl;

            }


        }

        void exibe_jogador(){

            cout << "| " << nome
            <<"|LVL| " << nivel
            << "HP | " << hp
            << "Lvl Alquimia | " << lvl_alquimia
            << "Lvl Combate  | " << lvl_combate
            << "Lvl HabGeral |" << lvl_habilidade_geral
            << "Lvl Magia    |" << lvl_magia
            << endl;

        }

};

int main(){

    Jogador jogador3("Ciri", 120, 450, 90, 40, 30, 80);
    jogador3.exibe_jogador();

    jogador3.recebe_cura(1000); 
    jogador3.exibe_jogador();

    Jogador jogador4("Dandelion", -10, -50, -5, -3, -2, -1);
    jogador4.exibe_jogador();


    return 0;

}